src/sentry/rules/processing/delayed_processing.py

```diff
@@ -219,5 +252,32 @@ def apply_delayed(project_id: int) -> DefaultDict[Rule, set[int]] | None:
         rules_to_fire = get_rules_to_fire(
             condition_group_results, rule_to_slow_conditions, rules_to_groups
         )
-        return rules_to_fire
-    return None
+    # Step 7: Ready, aim, fire!!
+    group_id_to_event = get_group_id_to_event(rulegroup_to_events, project)
+
+    now = timezone.now()
+    for rule, group_ids in rules_to_fire.items():
+        frequency = rule.data.get("frequency") or Rule.DEFAULT_FREQUENCY
+        freq_offset = now - timedelta(minutes=frequency)
+        groups = Group.objects.filter(id__in=group_ids)
+        for group in groups:
+            rule_statuses = bulk_get_rule_status(alert_rules, group, project)
+            status = rule_statuses[rule.id]
+            if status.last_active and status.last_active > freq_offset:
+                return
+
+            updated = (
+                GroupRuleStatus.objects.filter(id=status.id)
+                .exclude(last_active__gt=freq_offset)
+                .update(last_active=now)
+            )
+
+            if not updated:
+                return
+
+            event = group_id_to_event[group.id]
+            notification_uuid = str(uuid.uuid4())
+            rule_fire_history = history.record(rule, group, event.event_id, notification_uuid)
+            activate_downstream_actions(
```

Comment:
From the processor code, I think we need to safe_execute this? It seems the processor is saving the futures from `activate_downstream_actions` in an instance variable, and then executing them later
https://github.com/getsentry/sentry/blob/2ef9702056627c02a737c1452ad1e9c3d24bfd51/src/sentry/tasks/post_process.py#L1052-L1054